controllers/akodeploymentconfig/cluster/cluster_controller.go

```diff
@@ -54,6 +54,13 @@ type ClusterReconciler struct {
 	GetRemoteClient remote.ClusterClientGetter
 }
 
+// TODO: (adduarte remvoe) AKOO add a finalizer to cluster object ako-operator.networking.tkg.tanzu.vmware.com  which blocks cluster deletion, in case AKO needs to clean up resources in AVI Controller before cluster is deleted.  But some cases, AKO might just crashed or isn't even there, for example, cluster is not created successfully, AKOO will not remove the finalizer since AKO can't report the status correctly.
```

Comment:
We might also check if the ako package in the workload cluster is reconciled succeeded or not to determine if ako is functioning or not.
If ako is not working, we can also remove the finalizer as well since it will also block the cluster deletion.